### ISSUE TITLE
Automate tree-sitter update

### ISSUE THREAD

**Issue by ayoub-benali** [helix-editor/helix#5503]
We should automate the PRs that update the tree-sitter git references so that Helix is often up to date when it comes to language support.
This is already the case for nvim-treesitter via github action [here](https://github.com/nvim-treesitter/nvim-treesitter/blob/0ed09e2e6c39d9b27755f48bd8a407d3828367b3/.github/workflows/update-lockfile.yml). It create PRs like this [one](https://github.com/nvim-treesitter/nvim-treesitter/pull/4142).

It is something that I would be happy to do if there are no objections.

**Comment by the-mikedavis** [comment]
Tree-sitter repository update PRs aren't trivial to review: it's very easy to make a breaking change to the grammar that makes the current set of queries invalid (for example, renaming or removing a named node which is used in the queries) and tree-sitter grammar repositories are very rarely semantically versioned like crates dependencies. When reviewing these changes I look at the compare link between the two revisions and try to determine what changed in the grammar and how it will affect the queries. Using automated PRs for this puts the entire burden of that on reviewers instead of it being shared between contributor and reviewer, and there are already a lot of PRs waiting on review.

**Comment by ayoub-benali** [comment]
I can't speak about all tree-sitter repositories but for example the scala one has tests against regressions [here](https://github.com/tree-sitter/tree-sitter-scala/pull/125/files#diff-919ac210accac9ecc55a76d10a7590e3d85ca3f0e165b52d30f08faee486d0cbL1) and imagine other repositories have similar processes. 

I don't think any tree-sitter maintainer has any interest in introducing regressions, so most of the heavy lifting is done actually in the respective tree-sitter repository and not here. I guess there shouldn't be much to review since Helix is already relying on third parties for the syntax highlighting.

The goal is to lower the risk of issues being created here just because the tree-sitter grammar is out of date otherwise users have to maintain the git references for each languages them selves.

**Comment by archseer** [comment]
I agree, often there's structural changes that are easy to miss because they pass CI, but it would be nice to add them to highlighting queries.

For example if a language didn't support floats but added the grammar for it, a grammar bump would pass CI, but the highlighting wouldn't work until we realized it was added and manually introduce support.

**Comment by archseer** [comment]
A script that could list out the grammars that are out of date + show a diff would be nice though

**Comment by the-mikedavis** [comment]
> The goal is to lower the risk of issues being created here just because the tree-sitter grammar is out of date

This seems like premature automation to me: we see very few of these issues and they're fairly easy to fix.

**Comment by ayoub-benali** [comment]
> A script that could list out the grammars that are out of date + show a diff would be nice though

You mean the PR should update several languages and show the diff in the PR description ? 
Maybe one PR per languages is better ? This way we can close PRs that are breaking and merge the others
